core/src/main/java/com/oath/oak/NativeAllocator/OakNativeMemoryAllocator.java

```diff
@@ -93,11 +109,12 @@ public ByteBuffer allocate(int size) {
                 if ((blocks.size() + 1) * blocksProvider.blockSize() > capacity) {
                     throw new OakOutOfMemoryException();
                 } else {
+                    // going to allocate additional block (big chunk of memory)
+                    // need to be thread-safe, so not many blocks are allocated
+                    // locking is actually the most reasonable way of synchronization here
                     synchronized (this) {
```

Comment:
maybe better to synchronize on currentBlock

Reply:
fixed